Creation of enumeration reference for discrete math (#3092)
Codecademy/docs: content/discrete-math/concepts/enumeration/enumeration.md

---
Title: 'Enumeration'
Description: 'Enumeration is a complete, ordered listing of all the items in a collection.'
Subjects:
  - 'Computer Science'
  - 'Discrete Math'
Tags:
  - 'Arithmetic'
  - 'Math'
  - 'Logical'
  - 'Logic'
CatalogContent:
  - 'computer-science'
  - 'paths/computer-science'
---

> **Counting Theory** is a part of 'discrete mathematics' that deals with counting the number of possible outcomes of a given event. Being an essential element of 'probability', it tries to answer an abstract question, "How many?"
> 
> **Enumeration** is a complete, ordered listing of all the items in a collection. When speaking in the context of 'mathematics' and 'computer science', listing all the elements of a set comes into play.

Reviewer: ```suggestion
**Enumeration** is another word for count. In the context of mathematics, enumerating is a type of counting given a specified event or situation. An enumeration is a complete, ordered listing of all the items in a collection. When speaking in the context of mathematics and computer science, listing all the elements of a set comes into play.

**Counting Theory** is a part of discrete mathematics that deals with counting the number of possible outcomes of a given event. Being an essential element of probability, it tries to answer an abstract question, "How many?"
```

I think your definition for counting theory is spot on! Enumerating is synonymous with counting and so I think in the context of discrete math (the topic), that should be mentioned. Then, it would be good to go into counting theory.

## Fundamental Counting Principle

The fundamental counting principle states that the total number of outcomes of two or more independent events is the product of the number of outcomes of each individual event.

## Example

> Suppose you have 3 shirts (call them A , B , and C ), and 4 pairs of pants (call them w , x , y , and z). Then you have:

Reviewer 2: ```suggestion
Suppose there are 3 shirts (call them `A`, `B`, and `C`), and 4 pairs of pants (call them `w`, `x`, `y`, and `z`). There will be `3x4=12` possible outfits:
```

> 3×4=12

Reviewer 2: ```suggestion
```

> possible outfits:

Reviewer 2: ```suggestion
```

> 

Reviewer 2: Can we incase lines 33-35 in a [```plaintext code block](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#:~:text=Plaintext%3A%20%60%60%60plaintext)

Author: ayt

Noted with thanks

Author: did a commit; hope I got it right, hehe

```

## Principle of Inclusion and Exclusion

The principle of inclusion and exclusion is an approach that derives the method of finding the number of elements in the union of two (2) finite sets.

In the context of combinations and probability, the necessity of finding a counting method in order to make sure that an object is not counted twice is wanting.

> The formula:

Reviewer: ```suggestion
The formula (ꓴ is "AND", ꓵ is "OR"):
```

Reviewer: mentioning this once would be good

The formula:

> n(A ꓴ B) = n(A) + n(B) - n(A ꓵ B)

Reviewer 2: Incase line 45 into a ```tex code block.

Author: duly noted with thanks

Author: same here, hope I got it right

The exclusion of the intersection of the sets addresses the problem or question of duplication.

## Product Rule

> The rule states that if there are n(A) ways to do A and n(B) ways to do B, then the number of ways to do A nd B is n(A) x n(B). This is true if the number of ways of doing A and B are independent; the number of choices fordoing B is thesame regardless of which choice you made for A.

Reviewer 2: ```suggestion
The rule states that if there are `n(A)` ways to do `A` and `n(B)` ways to do `B`, then the number of ways to do A and B is `n(A) x n(B)`. This is true if the number of ways of doing `A` and `B` are independent; the number of choices for doing `B` is the same regardless of which choice is made for `A`.
```

The general take, however:

> There are n(A) x n(B) x n(C) ways to do A and B and C

Reviewer 2: ```suggestion
There are `n(A) x n(B) x n(C)` ways to do `A` and `B` and `C`.
```

## Sum Rule

> The rule states that if there are n(A) ways to do A and, distinct from them, n(B) ways to do B, then the number of ways to do A or B is n(A) + n(B)

Reviewer 2: ```suggestion
The rule states that if there are `n(A)` ways to do `A` and, distinct from them, `n(B)` ways to do `B`, then the number of ways to do `A` or `B` is `n(A) + n(B)`.
```

And, the general rule:

> There are n(A) + n(B) + n(C) ways to do A or B or C

Reviewer 2: ```suggestion
There are `n(A) + n(B) + n(C)` ways to do `A` or `B` or `C`.
```
